controllers/nnf_workflow_controller.go

```diff
@@ -120,22 +122,49 @@ func (r *NnfWorkflowReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			return ctrl.Result{}, nil
 		}
 
-		// TODO: Teardown workflow
-		/*
-			if err := r.teardownWorkflow(ctx, workflow); err != nil {
-				return ctrl.Result{}, err
-			}
+		// Run the teardown state logic in case the workflow was deleted without
```

Comment:
We can use the validation webhook in dws-operator to prevent a delete-without-teardown.  Add it to the ValidateDelete() method there.

Reply:
The workflow will be properly cleaned up even if it isn't in the teardown state, so there's no need to restrict delete to only when we're in teardown.